plugins/outputs/cratedb/cratedb.go

```diff
@@ -0,0 +1,202 @@
+package cratedb
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"sort"
+	"strings"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/plugins/outputs"
+	_ "github.com/jackc/pgx/stdlib"
+)
+
+type CrateDB struct {
+	URL         string
+	Timeout     internal.Duration
+	Table       string
+	TableCreate bool `toml:"table_create"`
+	DB          *sql.DB
+}
+
+var sampleConfig = `
+  # A lib/pq connection string.
+  # See http://godoc.org/github.com/lib/pq#hdr-Connection_String_Parameters
+  url = "postgres://user:password@localhost/schema?sslmode=disable"
+  # Timeout for all CrateDB queries.
+  timeout = "5s"
+  # Name of the table to store metrics in.
+  table = "metrics"
+  # If true, and the metrics table does not exist, create it automatically.
+  table_create = true
+`
+
+func (c *CrateDB) Connect() error {
+	db, err := sql.Open("pgx", c.URL)
+	if err != nil {
+		return err
+	} else if c.TableCreate {
+		sql := `
+CREATE TABLE IF NOT EXISTS ` + c.Table + ` (
+	"hash_id" LONG INDEX OFF,
+	"timestamp" TIMESTAMP,
+	"name" STRING,
+	"tags" OBJECT(DYNAMIC),
+	"fields" OBJECT(DYNAMIC),
+	"day" TIMESTAMP GENERATED ALWAYS AS date_trunc('day', "timestamp"),
+	PRIMARY KEY ("timestamp", "hash_id","day")
+) PARTITIONED BY("day");
+`
+		ctx, cancel := context.WithTimeout(context.Background(), c.Timeout.Duration)
+		defer cancel()
+		if _, err := db.ExecContext(ctx, sql); err != nil {
+			return err
+		}
+	}
+	c.DB = db
+	return nil
+}
+
+func (c *CrateDB) Write(metrics []telegraf.Metric) error {
+	ctx, cancel := context.WithTimeout(context.Background(), c.Timeout.Duration)
+	defer cancel()
+	if sql, err := insertSQL(c.Table, metrics, time.Local); err != nil {
+		return err
+	} else if _, err := c.DB.ExecContext(ctx, sql); err != nil {
+		return err
+	}
+	return nil
+}
+
+func insertSQL(table string, metrics []telegraf.Metric, loc *time.Location) (string, error) {
+	rows := make([]string, len(metrics))
+	for i, m := range metrics {
+		// Note: We have to convert HashID from uint64 to int64 below because
+		// CrateDB only supports a signed 64 bit LONG type which would give us
+		// problems, e.g.:
+		//
+		// CREATE TABLE my_long (val LONG);
+		// INSERT INTO my_long(val) VALUES (14305102049502225714);
+		// -> ERROR:  SQLParseException: For input string: "14305102049502225714"
+
+		cols := []interface{}{
+			int64(m.HashID()),
```

Comment:
I hadn't considered the values of this hash to be part stable across versions, would it be a problem if this hash changed even if the series key (measurement name + tagset) did not?

Reply:
Np. I think a cryptographic hash function makes more sense for the use case (deduplication) anyway. Let me know if you think https://github.com/influxdata/telegraf/pull/3210/commits/626f6532dbd2b5f7395e49d69d2ed8b9ee02b30d works for you.